pkg/distributor/distributor.go

```diff
@@ -334,6 +339,12 @@ func New(cfg Config, clientConfig ingester_client.Config, limits *validation.Ove
 		discardedExemplarsRateLimited:     validation.DiscardedExemplarsCounter(reg, validation.ReasonRateLimited),
 		discardedMetadataRateLimited:      validation.DiscardedMetadataCounter(reg, validation.ReasonRateLimited),
 
+		discardedInstanceRequests: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Namespace: "cortex",
+			Name:      "distributor_instance_discarded_requests_total",
+			Help:      "Requests discarded for hitting per-instance limits",
+		}, []string{"reason"}),
+
```

Comment:
Should we initialise the counter to 0 for each `reason` so that we can show it in dashboards even if no requests have been discarded?